tools/f18/CMakeLists.txt

```diff
@@ -36,4 +36,5 @@ add_executable(f18-parse-demo
 
 target_link_libraries(f18-parse-demo
   FortranParser
+  FortranEvaluate
```

Comment:
I think we could avoid this dependency by adding a dummy `GenericExprWrapper` to `f18-parse-demo.cc`.

Depending on evaluate doesn't seem right, but this cure may be worse than the disease.

Reply:
I tried that.  Will try again.

Reply:
Adding this to `f18-parse-demo.cc` before the other includes seems to work:
```
#include "../../lib/common/indirection.h"
namespace Fortran::evaluate {
struct GenericExprWrapper {
  bool operator==(const GenericExprWrapper &) const { return false; }
};
}
DEFINE_OWNING_DESTRUCTOR(OwningPointer, Fortran::evaluate::GenericExprWrapper)
```

Reply:
That's pretty much what I've discovered as well.  I'll take care of it.

Reply:
Maybe a cleaner approach would be to put the above in a .cc file in parser. Then the dummy version would hopefully be linked in if there wasn't a real version. So f18 would get the real one from evaluate and f18-parse-demo would get the dummy.

This would help other tools that need to depend only the parser, assuming there will be some. Otherwise they would each have to add their own dummy GenericExprWrapper.

Reply:
We don't want duplicate entry points in the f18 libraries.  I've defined a stub file for the tools that f18-parse-demo can use.  This is a temporary situation and it needs a better fix.  Please stand by, I'll push new code shortly.